libqtile/widget/redshift.py

```diff
@@ -0,0 +1,333 @@
+        (
+            "disabled_txt",
+            "󱠃",
+            "Text to show when redshift is disabled. NOTE: by default, a nerd icon is used."
+            "Available fields: 'brightness' brightness to set when redshift is enabled, "
+            "'is_enabled' boolean to state whether the widget is enabled or not, "
+            "'gamma_blue' gamma blue value to set when redshift is enabled, "
+            "'gamma_green' gamma green value to set when redshift is enabled, "
+            "'gamma_red' gamma red value to set when redshift is enabled, "
+            "'temperature' temperature to set when redshift is enabled, ",
+        ),
+        (
+            "enabled_txt",
+            "󰛨",
+            "Text to show when redshift is disabled. NOTE: by default, a nerd icon is used."
+            "Available fields: see disabled_txt's available fields for all of them",
+        ),
+        ("gamma_red", 1.0, "Redshift gamma red setting"),
+        ("gamma_blue", 1.0, "Redshift gamma blue"),
+        ("gamma_green", 1.0, "Redshift gamma green"),
+        ("font", "sans", "Default font"),
+        ("fontsize", 20, "Font size"),
+        ("foreground", "ffffff", "Font colour for information text"),
+        ("temperature", 1700, "Redshift temperature to set when enabled"),
+        (
+            "temperature_fmt",
+            "Temperature: {temperature}",
+            "Text to display when showing temperature text"
+            "Available fields: 'temperature' temperature to set when redshift is enabled. ",
+        ),
+        (
+            "brightness_fmt",
+            "Brightness: {brightness}",
+            "Text to display when showing brightness text"
+            "Available fields: 'brightness' brightness to set when redshift is enabled. ",
+        ),
+        (
+            "gamma_fmt",
+            "Gamma: {gamma_red}:{gamma_green}:{gamma_blue}",
+            "Text to display when showing gamma text"
+            "Available fields: 'gamma_red' gamma red value to set when redshift is enabled. "
+            "'gamma_blue' gamma blue value to set when redshift is enabled. "
+            "'gamma_green' gamma green value to set when redshift is enabled. ",
+        ),
+    ]
+
+    _dependencies = ["redshift"]
+
+    def __init__(self, **config):
+        _TextBox.__init__(self, **config)
+        self.add_defaults(Redshift.defaults)
+        self.add_defaults(base.PaddingMixin.defaults)
+        self.is_enabled = False
+        self._line_index = 0
+        self._lines = []
+        self.add_callbacks(
+            {
+                "Button1": self.click,
+                "Button4": self.scroll_up,
+                "Button5": self.scroll_down,
+            }
+        )
+
+    def _configure(self, qtile, bar):
+        _TextBox._configure(self, qtile, bar)
+        # disable redshift so we have a known initial state
+        self.reset_redshift()
+        self._set_lines()
+        self._set_text()
```

Comment:
```suggestion
        self.qtile.call_soon(self._set_text)
```
This will run `self._set_text()` once the bar's configuration has completed.